src/Certify.Shared/Management/PluginManager.cs

```diff
@@ -19,6 +19,12 @@ public class PluginManager
         public IDashboardClient DashboardClient { get; set; }
         public List<IDeploymentTaskProviderPlugin> DeploymentTaskProviders { get; set; }
         public List<ICertificateManagerProviderPlugin> CertificateManagerProviders { get; set; }
+        public List<IDnsProviderProviderPlugin> DnsProviderProviders { get; set; }
+        public bool DnsMSDNSFailedToLoad { get; private set; }
+
+        // ISSUE: Difficult to plumb PluginManager all the way through to ChallengeProviders, as that's about 7-8 layers deep on the call stack from where this is accessible.
+        // Since there's no injection container, making a cubbyhole for it here.
+        public static PluginManager _instance { get; private set; }
```

Comment:
`ChallengeProvider` needs a reference to `PluginManager` in order to get access to the challenge providers `PluginManager` loads. However, at the time `ChallengeProvider` is called into, the closest point it can get a `PluginManager` is in `CertifyManager` which is 7-8 frames up the call stack. Other places I've seen `ILog` passed around as method parameters, but in order to get things working quickly I just made it a `static`. If I knew how to use LightInject, that's definitely the better way to go.